quinn/tests/post_quantum.rs

```diff
@@ -0,0 +1,105 @@
+#![cfg(feature = "rustls-aws-lc-rs")]
+use proto::crypto::rustls::HandshakeData;
+use quinn::{
+    crypto::rustls::{QuicClientConfig, QuicServerConfig},
+    Endpoint,
+};
+use rustls::{
+    pki_types::{CertificateDer, PrivatePkcs8KeyDer},
+    NamedGroup,
+};
+
+use std::{
+    error::Error,
+    net::{Ipv4Addr, SocketAddr},
+    sync::Arc,
+};
+
+#[tokio::test]
+async fn post_quantum_key_exchange() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    rustls_post_quantum::provider().install_default().unwrap();
+    let server_addr = (Ipv4Addr::LOCALHOST, 8080).into();
+
+    let (endpoint, server_cert) = make_server_endpoint(server_addr)?;
+    // accept a single connection
+    tokio::spawn(async move {
+        let incoming_conn = endpoint.accept().await.unwrap();
+        let conn = incoming_conn.await.unwrap();
+        println!(
+            "[server] connection accepted: addr={}",
+            conn.remote_address()
+        );
+        assert_eq!(
+            conn.handshake_data()
+                .unwrap()
+                .downcast::<HandshakeData>()
+                .unwrap()
+                .negociated_key_exchange_group,
+            X25519_KYBER768_DRAFT00
+        )
+        // Dropping all handles associated with a connection implicitly closes it
+    });
+
+    let endpoint = make_client_endpoint((Ipv4Addr::UNSPECIFIED, 0).into(), &[&server_cert])?;
+    // connect to server
+    let connection = endpoint
+        .connect(server_addr, "localhost")
+        .unwrap()
+        .await
+        .unwrap();
+    println!("[client] connected: addr={}", connection.remote_address());
```

Comment:
Tracing.

Reply:
I've changed it. In general, do you prefer I close the thread or you do it?

Reply:
If you're reasonably sure you've addressed the feedback, feel free to resolve the thread.